### PR TITLE
Adds Links to EntitySet type Get response objects

### PR DESCRIPTION
**Pull request by irvinesunday**
Closes https://github.com/microsoftgraph/microsoft-graph-explorer-api/issues/185

This PR adds OpenAPI Link properties to EntitySet type Get response objects.

### CODE REVIEW

File: src/Microsoft.OpenApi.OData.Reader/Operation/EntitySetGetOperationHandler.cs [hunk 1]
```diff
@@ -192,7 +192,8 @@ protected override void SetResponses(OpenApiOperation operation)
                                     }
                                 }
                             }
-                        }
+                        },
+                        Links = Context.CreateLinks(EntitySet)
```
Comment (xuzhg) [inline src/Microsoft.OpenApi.OData.Reader/Operation/EntitySetGetOperationHandler.cs:196]: Do we have the similar for entity set, singleton get, navigation get?

Comment (darrelmiller) [reply src/Microsoft.OpenApi.OData.Reader/Operation/EntitySetGetOperationHandler.cs:196]: Wouldn't Singleton GET and Navigation GET be handled by the EntityGetOperationHandler?  If not, then yes we would need to add those too.

Comment (xuzhg) [reply src/Microsoft.OpenApi.OData.Reader/Operation/EntitySetGetOperationHandler.cs:196]: No, Singleton has its own SingletonGet handler. Same for Navigation property.

Comment (irvinesunday) [reply src/Microsoft.OpenApi.OData.Reader/Operation/EntitySetGetOperationHandler.cs]: Links have been updated for EntitySet, Singleton and Nagivation (GET)

File: src/Microsoft.OpenApi.OData.Reader/Generator/OpenApiLinkGenerator.cs [hunk 2]
```diff
@@ -21,22 +21,25 @@ internal static class OpenApiLinkGenerator
         /// Create the collection of <see cref="OpenApiLink"/> object.
         /// </summary>
         /// <param name="context">The OData context.</param>
-        /// <param name="entitySet">The Entity Set.</param>
+        /// <param name="entityType">The Entity type.</param>
+        /// <param name ="edmTypeName">The name of the EDM type.</param>
```
Comment (xuzhg) [inline src/Microsoft.OpenApi.OData.Reader/Generator/OpenApiLinkGenerator.cs:25]: >edmTypeName [](start = 26, length = 11)

The parameter name "edmTypeName" doesn't make sense. Can we name it "sourceName" or "navigationSourceName"?

File: test/Microsoft.OpenAPI.OData.Reader.Tests/Resources/TripService.OpenApi.yaml [hunk 3]
```diff
@@ -680,6 +693,19 @@ paths:
                     type: array
                     items:
                       $ref: '#/components/schemas/Microsoft.OData.Service.Sample.TrippinInMemory.Models.Person'
+          links:
+            Friends:
```
Comment (xuzhg) [inline test/Microsoft.OpenAPI.OData.Reader.Tests/Resources/TripService.OpenApi.yaml:697]: Did you run all the test cases? Make sure all the test cases can pass

**Comment by xuzhg** [comment]
@Irvine Sunday, please use "Squash and Merge" button to merge your PR.

File: test/Microsoft.OpenAPI.OData.Reader.Tests/Resources/TripService.OpenApi.yaml (hunk 3 above)
Comment (irvinesunday) [reply test/Microsoft.OpenAPI.OData.Reader.Tests/Resources/TripService.OpenApi.yaml:697]: Yes

File: src/Microsoft.OpenApi.OData.Reader/Generator/OpenApiLinkGenerator.cs (hunk 2 above)
Comment (irvinesunday) [reply src/Microsoft.OpenApi.OData.Reader/Generator/OpenApiLinkGenerator.cs:25]: sourceElementName

File: src/Microsoft.OpenApi.OData.Reader/Operation/EntityGetOperationHandler.cs [hunk 4]
```diff
@@ -99,7 +99,7 @@ protected override void SetResponses(OpenApiOperation operation)
                                 }
                             }
                         },
-                        Links = Context.CreateLinks(EntitySet)
+                        Links = Context.CreateLinks(EntitySet.EntityType(), EntitySet.Name)
```
Comment (darrelmiller) [inline src/Microsoft.OpenApi.OData.Reader/Operation/EntityGetOperationHandler.cs:102]: I think we should be using Entity name here not EntitySet Name

**Pull request merged**